AtSheet/explainer.md

```diff
@@ -0,0 +1,225 @@
+}
+</style>
+<template shadowrootmode="open">
+  <link rel="stylesheet" href="#foo" />
+  <span>I'm in the shadow DOM</span>
+</template>
+```
+or imported from JS:
+```html
+<script>
+  import {foo} from './sheet.css' with {type: 'css'};
+  ...
+  shadow.adoptedStyleSheets = [foo];
+</script>
+```
+
+## Detailed design discussion
+
+#### Named Imports with Imperative Shadow DOM
+
+`sheet.js` can also be imported via Javascript as follows:
+
+```js
+import baz, { bar } from 'sheet.css' with { type: 'css' }
+```
+
+`baz` will reference style rules outside of any `@sheet` blocks as a Default Import (in this case, the `div { color: blue; } ` rule).
+
+`bar` will reference style rules within the `@sheet bar` block as a Named Import (in this case, the `div { color: red; }  ` rule).
+
+Named imports may be renamed as part of this import process:
+
+```js
+import baz, { bar as renamed } from 'sheet.css' with { type: 'css' }
+```
+
+The default import may be omitted, importing only the named `@sheet`:
+
+```js
+import { bar } from 'sheet.css' with { type: 'css' }
+```
+
+Any of these `import` examples can be then used to set the `adoptedStyleSheets` attribute on a Shadow DOM node:
+
+```js
+import { bar } from 'sheet.css' with { type: 'css' }
+document.adoptedStyleSheets = [bar];
+shadowRoot.adoptedStyleSheets = [bar];
+```
+
+#### Performance
+
+This will be a performance-neutral feature, and use of it may allow for developers to reduce the number of network requests. We should ensure that multiple imports of different sheets from the same file produce a single network request. 
+
+```js
+// The following two imports should only make a single network request.
+import { foo } from 'sheet.css' with { type: 'css' };
+import { bar } from 'sheet.css' with { type: 'css' }
+```
+
+```html
+<style>
+/* The following two imports should only make a single network request. */
+@import "sheet.css#foo";
+@import "sheet.css#bar";
+</style>
+```
+
+```html
+<!-- The following two link tags should only make a single network request. -->
+<link rel="stylesheet" href="sheet.css#foo" />
+<link rel="stylesheet" href="sheet.css#bar" />
+```
+
+#### Interaction with CSSOM
+
+
+Named `@sheet` references augment the [existing](https://drafts.csswg.org/cssom/#stylesheet) `StyleSheet` interface with an optional `name` attribute reflecting the `@sheet` identifier:
+
+```
+[Exposed=Window]
+interface StyleSheet {
+  readonly attribute DOMString? name;
+};
+```
+*Open issue: Should this overload the existing `title` attribute instead?*
+
+This also expands the [existing](https://drafts.csswg.org/cssom/#cssstylesheet) CSSOM `CSSStyleSheet` definition with a `StyleSheetList` of nested `CSSStyleSheet` objects to access nested `@sheet` references:
+
+```
+[Exposed=Window]
+interface CSSStyleSheet : StyleSheet {
+  [SameObject] readonly attribute StyleSheetList nestedStyleSheets;
+};
+```
+*Open issue: The name `nestedStyleSheets` is up for discussion*
+
+## Considered alternatives
```

Comment:
Should we include all those mentioned in https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/ShadowDOM/explainer.md#alternate-proposals (minus `@sheet` as well?)

Reply:
I just gave a reference. Can copy over if it's something that other folks trip over.